### ISSUE TITLE
Conflict with js-yaml? 

### ISSUE THREAD

**Issue by megastef** [connec/yaml-js#37]
Hi, 

does 'yaml-js' modify any global object, which could cause a conflict with 'js-yaml'? 

We use js-yaml, but one of the plugins did use yaml-js (via rc-yaml) and then we got the problem that regular expression in our config could not be parsed (a feature of js-yaml). 

I was able to solve the conflict, by changing rc-yaml to use js-yaml (publihed as rc-yaml-2 on npm) - but I have no idea how the two node modules could make trouble in another one. 

Here is the error message we found initially:
```
{ could not determine a constructor for the tag tag:yaml.org,2002:js/regexp
  on line 22, column 20
    at ConstructorError.YAMLError [as constructor] (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/errors.js:70:46)
    at ConstructorError.MarkedYAMLError [as constructor] (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/errors.js:90:45)
    at new ConstructorError (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/constructor.js:22:53)
    at Loader.__dirname.Constructor.Constructor.construct_undefined (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/constructor.js:629:13)
    at Loader.__dirname.BaseConstructor.BaseConstructor.construct_object (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/constructor.js:208:28)
    at Loader.__dirname.BaseConstructor.BaseConstructor.construct_mapping (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/constructor.js:248:22)
    at Loader.__dirname.Constructor.Constructor.construct_mapping (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/constructor.js:368:54)
    at /usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/constructor.js:598:23
    at Loader.__dirname.BaseConstructor.BaseConstructor.construct_document (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/constructor.js:163:41)
    at Loader.__dirname.BaseConstructor.BaseConstructor.get_single_data (/usr/lib/node_modules/@sematext/logagent-nodejs-monitor/node_modules/yaml-js/lib/constructor.js:147:21)
  context: null,
  context_mark: null,
  problem: 'could not determine a constructor for the tag tag:yaml.org,2002:js/regexp',
  problem_mark: 
   Mark {
     line: 21,
     column: 19,
```

The workaround was to replace in the sub-modules the dependency to yaml-js with js-yaml and our program did work again.

**Comment by connec** [comment]
This is likely due to `yaml-js` injecting a `require.extensions` handler for YAML. This is pretty bad practice now, so I'll take that out and update.

Thanks for the report!

**Comment by shockey** [comment]
@connec, any updates on removing the `require.extensions` from this module?

We use the library in `swagger-ui`, and the `require.extensions` code seems to be causing some problems for us: https://github.com/swagger-api/swagger-ui/issues/3046

**Comment by connec** [comment]
Fixed in 20c58d85e1d778a11e021a40f661717c6498ef3d, published as `0.2.0`!

**Comment by shockey** [comment]
@connec, this seems to be working great. Thanks for the help!
